snap: migrate to base: core22

Saviq opened the pull request.

Saviq committed:
snap: refresh for `base: core22`

Saviq committed:
tmp: use mesa-core22-test for now

Saviq committed:
snap: drop old workaround

Saviq committed:
snap: drop `recipe-version` part

Saviq committed:
snap: migrate to CRAFT_ variables

Saviq committed:
ci: publish on the `22` track

Saviq committed:
snap: migrate to new graphics-core22 design

Saviq committed:
ci: add multiarch

AlanGriffiths commented on snap/hooks/connect-plug-graphics-core22:
I thought we'd dropped this "hooks" approach?

Saviq committed:
snap: drop libdrm

Saviq committed:
snap: drop hooks

Saviq committed:
snap: mesa-core22/latest/edge

Saviq committed:
ci: drop multiarch due to bug in action

https://github.com/diddlesnaps/snapcraft-multiarch-action/issues/22

Saviq committed:
snap: move to mesa-core22 stable

Saviq committed:
wrapper: drop the extra is-connected check

Saviq committed:
snap: prefer symlinks over binds

They're much cheaper, and binds are only necessary if the path already
exists in the base snap.

Saviq committed:
ci: close-snap for 22/, too

AlanGriffiths reviewed (approved):
bors merge
